app/src/main/java/com/yalta/repositories/RealRepo.kt

```diff
@@ -19,31 +20,31 @@ open class RealRepo {
                     .parse("text/plain"), body))
             .build()
         return try {
-            client.newCall(request).await()
+            Either.Right(client.newCall(request).await())
         } catch (e: Exception) {
-            when (e) {
-                is SocketTimeoutException, is ConnectException -> null
-                else -> throw e
-            }
+            Either.Left(when (e) {
+                is SocketTimeoutException, is ConnectException -> Reason.FAILED_CONNECTION
+                else -> Reason.UNKNOWN
```

Comment:
I'd log exception here at least

Reply:
There are no logs in the whole app, and I don't want to add logging here
You've also proposed to add AlertDialog - that's really not easy, because it's actually a fragment which can be started only in UI, so I need to pass the exception to UI, and that's complicated... Please, don't do this to me(

Reply:
Ok, too much of mess, I get it